rosidl_typesupport_opensplice_c/resource/msg__type_support_c.cpp.template

```diff
@@ -69,30 +69,34 @@ ROSIDL_GET_TYPE_SUPPORT_FUNCTION(@(pkg), @(subfolder), @(msg))();
 have_not_included_primitive_arrays = True
 have_not_included_string = True
 }@
-@[for field in spec.fields]@
-@[if field.type.is_primitive_type()]@
-@[if field.type.is_array and have_not_included_primitive_arrays]@
+{
+from collections import OrderedSet
+field_types = OrderedSet(sorted([field.type for field in spec.fields]))
```

Comment:
Instead of using an ordered set here I would suggest to just collect them in a plain set. Later the include directives should be ordered based on the string value of the included headers to comply with the linter.